tests/test_parsing.py

```diff
@@ -22,4 +25,92 @@ def test_format_version(self):
         """Test that the format version is correct."""
         assert self.hoa_header.format_version == "v1"
 
-    # TODO many other tests...
+    def test_nb_states(self):
+        """Test that the number of states is correct."""
+        assert self.hoa_header.nb_states == 2
+
+    def test_start_states(self):
+        """Test that the initial states are correct."""
+        assert self.hoa_header.start_states == [0]
+
+    def test_acc_name(self):
+        """Test that the acc-name is correct."""
+        assert self.hoa_header.acceptance_name == "Rabin 1"
+
+    def test_acceptance(self):
+        """Test that the acceptance condition is correct."""
+        assert self.hoa_header.acceptance.condition.nb_accepting_sets == 2
+        assert self.hoa_header.acceptance == Acceptance(And([Atom(0, AtomType.FINITE), Atom(1, AtomType.INFINITE),
+                                                             TrueAcceptance()]))
+
+    def test_AP(self):
+        """Test that AP is correct."""
+        assert self.hoa_header.propositions == ('a', 'b')
+        assert len(self.hoa_header.propositions) == 2
+
+    def test_aliases(self):
+        """Test that Alias is correct."""
+        assert self.hoa_header.aliases[0] == AliasLabelExpression('@a', AtomLabelExpression("0"))
+        assert self.hoa_header.aliases[1] == AliasLabelExpression('@b', AtomLabelExpression("1"))
+
+    def test_body(self):
+        """Test that the HOA body is correct."""
+        state_edges_dict = OrderedDict({})
+        state_edges_dict[State(0, None, '"a U b"', None)] = [Edge([0], AndLabelExpression([AtomLabelExpression("0"),
```

Comment:
For initialize `State` I'd prefer the following:

`State(index=0, name='"a U b")`

The same applies for all occurrences of such type of constructors (i.e. with many optional keyworded arguments.)

Reply:
Also: `'"a U b"'` sounds like a bug in the parser. It should be quoted only once!

Reply:
> For initialize `State` I'd prefer the following:
> 
> `State(index=0, name='"a U b")`
> 
> The same applies for all occurrences of such type of constructors (i.e. with many optional keyworded arguments.)

ok, i'll change them

Reply:
> Also: `'"a U b"'` sounds like a bug in the parser. It should be quoted only once!

I left it as is because I thought it was right to parse also double quotes

Reply:
I think it should be not double quoted. However, we must remember it when we dump that field in a file, since the quote is required.